### PR TITLE
Add status screen UI and logic

### PR DESCRIPTION
**Pull request by GuyNoimark**
image

### CODE REVIEW

**Review by Ido-Sobol — COMMENTED** [review]
Nice!

File: lib/models/helpers.dart [hunk 1]
```diff
@@ -11,6 +13,18 @@ extension Cast<E extends num> on Iterable<E> {
   }
 }
 
+RobotMatchStatus titleToEnum(final String title) {
```
Comment (giladwo) [inline lib/models/helpers.dart:16]: This is a smell.
You shouldn't ever need to go this way - only generate displayable widgets/strings from internal typed representation i.e. enums.
If you insist on having enums here *in addition* to the db, then they should be constructed from the `id`s of the suitable rows on query/subscription.
The path is `gql -> enum -> widget` (no arrows going backwards - if the widget needs to know what value it's related to it should get it on construction etc.).

**Comment by giladwo** [comment]
There's way too much going on here.
Better split to more manageable change sets next time, to help reviewers comprehend your proposals.

**Comment by giladwo** [comment]
Oh wait, it's @itamarsch's force-pushing stuff around.
First, let @GuyNoimark handle his own PRs.
Second, this merge creates more friction that actually helping.

What you (Guy) should do here is:
```bash
git fetch --all -p
git switch origin/master -c feature/status-screen
git cherry-pick 8bdbc9c5 cd40ca33  # these are your two commits, status_screen^N and status_screen^N~ where N is either 1 or 2
git push -u origin feature/status-screen
```
(this can also be easily fixed with a `rebase`, but I can't give you a fool-proof recipe this easily).

**Comment by giladwo** [comment]
Wait, is this an older version of #246 ?
If not, open a clean PR after the above fix (or equivalent).

**Comment by itamarsch** [comment]
> Wait, is this an older version of #246 ?
> 
> If not, open a clean PR after the above fix (or equivalent).

This is an older version

**Pull request closed without merging**